backend-project/small_eod/institutions/serializers.py

```diff
@@ -69,8 +72,24 @@ def create(self, validated_data):
         validated_data["address"] = AddressData.objects.create(
             **validated_data.pop("address")
         )
-        validated_data["administrative_unit"] = JednostkaAdministracyjna.objects.create(
-            **validated_data.pop("administrative_unit")
-        )
+        administrative_unit = validated_data.pop("administrative_unit")
+        institution = super().create(validated_data)
+        institution.administrative_unit = administrative_unit
+        institution.save()
+        return institution
+
+    def update(self, instance, validated_data):
+
+        institution_nested = [
+            {
+                "instance": instance.external_identifier,
+                "data": validated_data.pop("external_identifier", {}),
+            },
+            {"instance": instance.address, "data": validated_data.pop("address", {})},
+        ]
 
-        return super().create(validated_data)
+        for nested_object in institution_nested:
```

Comment:
Podoba mi się, że tak zgrabnie to zapisano, ale mam wrażenie, że przydałby się tu krótki komentarz co chcesz osiągnąć, bo kod wymaga chwili na zrozumienie. Drobiazg.

Reply:
Zrobione